src/Microsoft.AspNetCore.Authorization/IAuthorizationEvaluator.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Authorization
+{
+    /// <summary>
+    /// Determines whether an authorization request was successful or not.
+    /// </summary>
+    public interface IAuthorizationEvaluator
+    {
+        /// <summary>
+        /// Returns true, if authorization has failed.
+        /// </summary>
+        /// <param name="context">The authorization information.</param>
+        /// <returns>True if authorization has failed.</returns>
+        bool HasFailed(AuthorizationHandlerContext context);
```

Comment:
HasFailed is there for short circuiting. Both can be false since succeeded is only true if all requirements are met by default. The default implementation does not short circuit executing all handlers but that was an explicit ask to be enabled.